server/src/BitBakeProjectScanner.ts

```diff
@@ -322,80 +300,71 @@ export class BitBakeProjectScanner {
-            if (match.index === regExp.lastIndex) {
-              regExp.lastIndex++
-            }
 
-            recipeWithOutPath.path = path.parse(match[0].trim())
-          }
+        for (const match of output.matchAll(regExp)) {
+          recipeWithOutPath.path = path.parse(match[0].trim())
         }
       }
     }
   }
 
   private scanRecipesAppends (): void {
-    const output: string = this.executeBitBakeCommand('bitbake-layers show-appends')
+    const commandResult = this.executeBitBakeCommand('bitbake-layers show-appends')
+    const output = commandResult.output.toString()
 
-    if (output.length > 0) {
-      const outerReg: RegExp = /(\S.*\.bb):(?:\s*\/\S*.bbappend)+/g
+    const outerReg: RegExp = /(\S.*\.bb):(?:\s*\/\S*.bbappend)+/g
 
-      let match: RegExpExecArray | null
+    for (const match of output.matchAll(outerReg)) {
+      const fullRecipeNameAsArray: string[] = match[1].split('_')
 
-      while ((match = outerReg.exec(output)) !== null) {
-        if (match.index === outerReg.lastIndex) {
-          outerReg.lastIndex++
-        }
-        let matchInner: RegExpExecArray | null
-        const fullRecipeNameAsArray: string[] = match[1].split('_')
-
-        if (fullRecipeNameAsArray.length > 0) {
-          const recipeName: string = fullRecipeNameAsArray[0]
-
-          const recipe: ElementInfo | undefined = this.recipes.find((obj: ElementInfo): boolean => {
-            return obj.name === recipeName
-          })
+      if (fullRecipeNameAsArray.length > 0) {
+        const recipeName: string = fullRecipeNameAsArray[0]
 
-          if (recipe !== undefined) {
-            const innerReg: RegExp = /(\S*\.bbappend)/g
-
-            while ((matchInner = innerReg.exec(match[0])) !== null) {
-              if (matchInner.index === innerReg.lastIndex) {
-                innerReg.lastIndex++
-              }
+        const recipe: ElementInfo | undefined = this.recipes.find((obj: ElementInfo): boolean => {
+          return obj.name === recipeName
+        })
 
-              if (recipe.appends === undefined) {
-                recipe.appends = new Array < PathInfo >()
-              }
+        if (recipe !== undefined) {
+          const innerReg: RegExp = /(\S*\.bbappend)/g
 
-              recipe.appends.push(path.parse(matchInner[0]))
+          for (const matchInner of match[0].matchAll(innerReg)) {
+            if (recipe.appends === undefined) {
+              recipe.appends = new Array < PathInfo >()
             }
+
+            recipe.appends.push(path.parse(matchInner[0]))
           }
         }
       }
     }
   }
 
-  private executeBitBakeCommand (command: string): string {
+  private executeBitBakeCommand (command: string): childProcess.SpawnSyncReturns<Buffer> {
     const scriptContent: string = this.generateBitBakeCommand(command)
-    return this.executeCommand(scriptContent)
+    const commandResult = this.executeCommand(scriptContent)
+
+    if (commandResult.status === 127) {
```

Comment:
See my comment on https://github.com/savoirfairelinux/vscode-bitbake/pull/13
Error handling should be done at the caller level taht can handle the error. Splitting error cases like this will make the code very confusing IMO.

Reply:
It's hard to read the code when you are changing the essential primitive function like this one in multiple commits. Maybe try to have one commit that adds this library function properly, add the features around it.